orttraining/orttraining/core/framework/module_gradient_graph_builder.cc

```diff
@@ -50,7 +50,10 @@ Status ModuleGradientGraphBuilder::Initialize(std::istream& model_istream,
 
   // Remove the training initializers from the graph and move them to graph inputs.
   for (const auto& initializer_name : training_graph_info_.initializer_names_to_train) {
-    input_args.emplace_back(graph.GetNodeArg(initializer_name));
+    const NodeArg* node_arg = graph.GetNodeArg(initializer_name);
+    if (node_arg) {
```

Comment:
maybe replace the if  check with ORT_ENFORCE(node_arg != nullptr) ?

Reply:
If we have an unused layer such as the following defined in the module __init__ function the corresponding node_arg is zero at this point. 
        self.fc2 = torch.nn.Linear(hidden_size, num_classes)
Should we still keep the assertion?